addons/web/static/src/core/notebook/notebook.js

```diff
@@ -0,0 +1,156 @@
+/** @odoo-module **/
+
+import { scrollTo } from "@web/core/utils/scrolling";
+
+const { Component, onWillUpdateProps, useEffect, useRef, useState } = owl;
+
+/**
+ * A notebook component that will render only the current page and allow
+ * switching between its pages.
+ *
+ * You can also set pages using a template component. Use an array with
+ * the `pages` props to do such rendering.
+ *
+ * Pages can also specify their index in the notebook.
+ *
+ *      e.g.:
+ *          PageTemplate.template = xml`
+                    <h1 t-esc="props.title" />
+                    <p t-esc="props.text" />`;
+
+ *      `pages` could be:
+ *      [
+ *          {
+ *              Component: PageTemplate,
+ *              props: {
+ *                  title: "Some Title 1",
+ *                  isVisible: bool,
+ *                  text: "Text Content 1",
+ *              },
+ *              name: 'some_name' // optional
+ *          },
+ *          {
+ *              Component: PageTemplate,
+ *              props: {
+ *                  title: "Some Title 2",
+ *                  isVisible: bool,
+ *                  text: "Text Content 2",
+ *              },
+ *          },
+ *      ]
+ *
+ * <Notebook pages="pages">
+ *    <t t-set-slot="Page Name 1" title="Some Title" isVisible="bool">
+ *      <div>Page Content 1</div>
+ *    </t>
+ *    <t t-set-slot="Page Name 2" title="Some Title" isVisible="bool">
+ *      <div>Page Content 2</div>
+ *    </t>
+ * </Notebook>
+ *
+ * @extends Component
+ */
+
+export class Notebook extends Component {
+    setup() {
+        this.activePane = useRef("activePane");
+        this.anchorTarget = null;
+        this.pages = this.computePages(this.props);
+        this.state = useState({ currentPage: null });
+        this.state.currentPage = this.props.defaultPage || this.computeActivePage();
+        this.env.bus.addEventListener("SCROLLER:ANCHOR_LINK_CLICKED", (ev) =>
```

Comment:
it looks like this handler is not unregistered anywhere? this is a memory leak then